classes/conversion.php

```diff
@@ -963,32 +964,41 @@ private function get_transcode_files(\stdClass $conversionrecord, $handler=null)
                 'filepath' => '/' . $conversionrecord->contenthash . '/conversions/',
                 'filename' => $filename,
             );
-
             $downloadparams = array(
                 'Bucket' => $this->config->s3_output_bucket, // Required.
                 'Key' => $availableobject['Key'], // Required.
             );
 
-            $getobject = $s3client->getObject($downloadparams);
-            $filecontent = $getobject->get('Body');
+            try {
+                // The playlist files (including iframe playlists) created in s3 transcoding contain
+                // relative file paths to Variant Streams for adaptive bitsteaming media, these need to be amended
+                // to have Moodle plugin filepaths.
+                if ($this->is_file_playlist($filename)) {
+                    // This is small enough that we can just pull direct to memory.
+                    $result = $s3client->getObject($downloadparams);
+                    $filecontent = $result->get('Body');
+                    $filecontent = $this->replace_playlist_urls_with_pluginfile_urls($filecontent, $conversionrecord->contenthash);
 
-            // The playlist files (including iframe playlists) created in s3 transcoding contain
-            // relative file paths to Variant Streams for adaptive bitsteaming media, these need to be amended
-            // to have Moodle plugin filepaths.
-            if ($this->is_file_playlist($filename)) {
-                $filecontent = $this->replace_playlist_urls_with_pluginfile_urls($filecontent, $conversionrecord->contenthash);
-            }
+                    $transcodedfile = $fs->create_file_from_string($filerecord, $filecontent);
 
-            try {
-                $trancodedfile = $fs->create_file_from_string($filerecord, $filecontent);
+                } else {
+                    // Video file handling. Might be too big for memory, write to disk first.
+                    $filetarget = $requestdir . '/' . $filename;
+                    $downloadparams['SaveAs'] = $filetarget;
+                    $s3client->getObject($downloadparams);
+
+                    $transcodedfile = $fs->create_file_from_pathname($filerecord, $filetarget);
+                }
             } catch (\moodle_exception $e) {
                 // This file may already exist. Perhaps a reprocessed queue message.
                 // Either way, there isn't anything we can do about it.
                 // Move on.
                 continue;
             }
-
-            $transcodedfiles[] = $trancodedfile;
+            if (file_exists($filetarget)) {
+                unlink($filetarget);
+            }
```

Comment:
The variable only exists in this block above: https://github.com/catalyst/moodle-local_smartmedia/blob/c1dfd7b1683eabec03c17c3a4bfbfe58a48425bd/classes/conversion.php#L984-L988

Perhaps consider moving it there instead.

Reply:
Ah yeah this is a leftover from a small logic flow change. Nice catch